app/models/extra_mobile_data_request.rb

```diff
@@ -102,6 +105,11 @@ def validate_school_or_rb_present
     end
   end
 
+  def validate_request_uniqueness
+    scope = school_id.present? ? { school_id: school_id } : { responsible_body_id: responsible_body_id }
+    errors.add(:device_phone_number, :duplicate) if has_already_been_made?(scope)
```

Comment:
I'm not sure testing duplicates purely on phone number is right.

We don't allow users to edit requests, so if they make a mistake – eg wrong network – their only route is to do a new request – but this route is now being cut off.

My understanding of duplicate was same number, name and network.

Reply:
@fofr `has_already_been_made?` checks that they have the same number, network, name and RB/school